File: Frontend/src/services/presenceService.ts
export type PresenceRecord = {
  epc: string;
  lastSeen: string;
  antennaId: number;
  antennaName: string;
  zone: string;
};

class PresenceService {
  private readonly storageKey =
    "rfid-presence";

  registerRead(
    epc: string,
    timestamp: string,
    antennaId: number,
    antennaName: string,
    zone: string
  )
  {
    const records =
      this.getRecords();

    records[epc] = {
      epc,
      lastSeen: timestamp,
      antennaId,
      antennaName,
      zone,
    };

    localStorage.setItem(
      this.storageKey,
      JSON.stringify(records)
    );
  }

  private getRecords(): Record<
    string,
    PresenceRecord
  > {
    return JSON.parse(
      localStorage.getItem(
        this.storageKey
      ) ?? "{}"
    );
  }

  getLastSeen(
    epc: string
  ) {
    const records =
      this.getRecords();

    return records[epc];
  }

  isPresent(
    epc: string
  ) {
    const record =
      this.getLastSeen(epc);

    if (!record) {
      return false;
    }

    const lastSeen =
      new Date(
        record.lastSeen
      ).getTime();

    const now =
      Date.now();

    return (
      now - lastSeen <
      60000
    );
  }

  getAll() {
    return Object.values(
      this.getRecords()
    );
  }
}

export const presenceService =
  new PresenceService();

File: RFIDFlow.API/Services/PresenceService.cs
using RFIDFlow.API.Models;

namespace RFIDFlow.API.Services;

public class PresenceService
{
    private readonly Dictionary<
        string,
        PresenceRecord
    > _presence = new();

    public void RegisterRead(
        string epc)
    {
        _presence[epc] =
            new PresenceRecord
            {
                EPC = epc,
                LastSeen =
                    DateTime.UtcNow,
                Present = true
            };
    }

    public List<PresenceRecord>
        GetPresence()
    {
        foreach (
            var record in
            _presence.Values
        )
        {
            record.Present = true;
        }

        return _presence
            .Values
            .ToList();
    }
}